File: banking-domain/frontend/banking-ui/src/app/banking.models.ts
export interface BankingSnapshot {
  customerName: string;
  totalBalance: number;
  investedAssets: number;
  monthlyIncome: number;
  monthlySpend: number;
  creditUtilization: number;
  accounts: AccountSummary[];
  recentTransactions: TransactionItem[];
  spendingCategories: SpendingCategory[];
  marketMovers: MarketMover[];
  alerts: RiskAlert[];
}

export interface AccountSummary {
  id: number;
  name: string;
  type: string;
  numberMasked: string;
  balance: number;
  changePercent: number;
  status: string;
}

export interface TransactionItem {
  id: number;
  date: string;
  description: string;
  category: string;
  account: string;
  amount: number;
  direction: 'credit' | 'debit';
  status: string;
}

export interface TransactionInput {
  date: string;
  description: string;
  category: string;
  account: string;
  amount: number;
  direction: 'credit' | 'debit';
  status: string;
}

export interface SpendingCategory {
  label: string;
  amount: number;
  percentage: number;
}

export interface MarketMover {
  symbol: string;
  name: string;
  price: number;
  changePercent: number;
}

export interface RiskAlert {
  title: string;
  severity: string;
  detail: string;
}

export interface ApiSession {
  authenticationEnabled: boolean;
  isAuthenticated: boolean;
  name: string | null;
  subject: string | null;
  authenticationType: string | null;
}

File: banking-domain/backend/BankingApi/Models/BankingModels.cs
namespace NorthstarBankApi.Models;

public record BankingSnapshot(
    string CustomerName,
    decimal TotalBalance,
    decimal InvestedAssets,
    decimal MonthlyIncome,
    decimal MonthlySpend,
    decimal CreditUtilization,
    IReadOnlyList<AccountSummary> Accounts,
    IReadOnlyList<TransactionItem> RecentTransactions,
    IReadOnlyList<SpendingCategory> SpendingCategories,
    IReadOnlyList<MarketMover> MarketMovers,
    IReadOnlyList<RiskAlert> Alerts);

public record AccountSummary(
    int Id,
    string Name,
    string Type,
    string NumberMasked,
    decimal Balance,
    decimal ChangePercent,
    string Status);

public record TransactionItem(
    int Id,
    string Date,
    string Description,
    string Category,
    string Account,
    decimal Amount,
    string Direction,
    string Status);

public record UpsertTransactionRequest(
    string Date,
    string Description,
    string Category,
    string Account,
    decimal Amount,
    string Direction,
    string Status);

public record SpendingCategory(
    string Label,
    decimal Amount,
    decimal Percentage);

public record MarketMover(
    string Symbol,
    string Name,
    decimal Price,
    decimal ChangePercent);

public record RiskAlert(
    string Title,
    string Severity,
    string Detail);

public record ApiSession(
    bool AuthenticationEnabled,
    bool IsAuthenticated,
    string? Name,
    string? Subject,
    string? AuthenticationType);
